easy_biologic/program.py

```diff
@@ -645,43 +666,66 @@ def _save_data_together(
                         logging.warning( f'Error writing header: {err}' )
 
                 # write data
-                # get maximum rows
-                num_rows = { ch: len( data ) for ch, data in self._unsaved_data.items() }
-                written = { ch: [] for ch in self._unsaved_data.keys() }
-                for index in range( max( num_rows.values() ) ):
-                    written_row = { ch: None for ch in self._unsaved_data.keys() }
-                    row_data = ''
+                if pandas_installed:

```

Comment:
It may be worth breaking these into their own functions for readability. e.g. `write_with_pandas` and `write_with_default`.